guava/src/com/google/common/base/Ascii.java

```diff
@@ -629,4 +630,155 @@ private static int getAlphaIndex(char c) {
     // Fold upper-case ASCII to lower-case and make zero-indexed and unsigned (by casting to char).
     return (char) ((c | 0x20) - 'a');
   }
+
+  /**
+   * Searches through {@code source} to find {@code target}, ignoring the case of
+   * any ASCII alphabetic characters between {@code 'a'} and {@code 'z'}
+   * or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @param source       the characters being searched.
+   * @param sourceOffset offset of the source string.
+   * @param sourceCount  count of the source string.
+   * @param target       the characters being searched for.
+   * @param targetOffset offset of the target string.
+   * @param targetCount  count of the target string.
+   * @param fromIndex    the index to begin searching from.
+   */
+  private static int indexOfIgnoreCase(char[] source, int sourceOffset, int sourceCount,
```

Comment:
You can copy the whole method, changing two char array arguments to char sequence arguments so that you can deal with strings and string builders directly without `toCharArray` calls that duplicate all the char contents.